### PR TITLE
chore: minor refactor get datasource from action

### PR DESCRIPTION
**Pull request by nsarupr**

### CODE REVIEW

File: app/server/appsmith-server/src/main/java/com/appsmith/server/services/ce/DatasourceServiceCEImpl.java [hunk 1]
```diff
@@ -182,6 +182,11 @@ public Mono<Datasource> populateHintMessages(Datasource datasource) {
                 });
     }
 
+    @Override
+    public Mono<Datasource> getValidDatasourceMono(Datasource datasource) {
```
Comment (trishaanand) [inline app/server/appsmith-server/src/main/java/com/appsmith/server/services/ce/DatasourceServiceCEImpl.java:186]:
```suggestion
    public Mono<Datasource> getValidDatasourceFromActionMono(ActionDTO action, AclPermission aclPermission) {
```

File: app/server/appsmith-server/src/main/java/com/appsmith/server/services/ce/DatasourceServiceCEImpl.java [hunk 2]
```diff
@@ -182,6 +182,11 @@ public Mono<Datasource> populateHintMessages(Datasource datasource) {
...
+        return Mono.just(datasource);
```
Comment (trishaanand) [inline app/server/appsmith-server/src/main/java/com/appsmith/server/services/ce/DatasourceServiceCEImpl.java:187]:
```suggestion
        // Global datasource requires us to fetch the datasource from DB.
                    if (action.getDatasource() != null && action.getDatasource().getId() != null) {
                        return datasourceService.findById(action.getDatasource().getId(), aclPermission)
                                .switchIfEmpty(Mono.error(new AppsmithException(AppsmithError.NO_RESOURCE_FOUND,
                                        FieldName.DATASOURCE,
                                        action.getDatasource().getId())));
                    }
                    // This is a nested datasource. Return as is.
                    return Mono.justOrEmpty(action.getDatasource());
```

**Review by trishaanand — COMMENTED** [review]
What do you think about something like this?

File: app/server/appsmith-server/src/main/java/com/appsmith/server/services/ce/DatasourceServiceCEImpl.java (hunk 2 above)
Comment (nsarupr) [reply app/server/appsmith-server/src/main/java/com/appsmith/server/services/ce/DatasourceServiceCEImpl.java:187]: yeah, this makes sense. 
we have just moved this code from `NewActionServiceCEImpl`  ->  `DatasourceServiceCEImpl`
for better usage.

**Review by trishaanand — APPROVED** [review]

**Pull request merged**